lib/Sema/CSDiagnostics.cpp

```diff
@@ -520,10 +520,11 @@ bool MissingConformanceFailure::diagnoseAsAmbiguousOperatorRef() {
   // about missing conformance just in case.
   auto operatorID = name.getIdentifier();
 
-  auto *applyExpr = cast<ApplyExpr>(findParentExpr(anchor));
-  if (auto *binaryOp = dyn_cast<BinaryExpr>(applyExpr)) {
-    auto lhsType = getType(binaryOp->getArg()->getElement(0));
-    auto rhsType = getType(binaryOp->getArg()->getElement(1));
+  auto *fnType = getType(anchor)->getAs<AnyFunctionType>();
+  auto params = fnType->getParams();
+  if (params.size() == 2) {
```

Comment:
I think we have to figure out whether picked declaration for this case is actually an operator before we can produce diagnostics about operators because otherwise e.g.:

```swift
func myAssertion<T>(_ a: T, _ op: ((T,T)->Bool), _ b: T) {}

func foo(_: Int, _: Int) {}
func foo<T: BinaryInteger>(_: T, _: T) {}
// etc.

var e: Error? = nil
myAssertion(e, foo, nil)
```

Is going to produce a diagnostic about binary operator, although `foo` is obviously not one...

Reply:
We do, we check that the name is an operator a few lines above this

Reply:
```cpp
  if (!(name.isOperator() && getLHS()->isStdlibType() && getRHS()->isStdlibType()))
    return false;
```

Reply:
Good then :)